src/whitelist_policy.hpp

```diff
@@ -0,0 +1,66 @@
+/*
+  Copyright (c) 2014-2015 DataStax
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+#ifndef __CASS_WHITELIST_POLICY_HPP_INCLUDED__
+#define __CASS_WHITELIST_POLICY_HPP_INCLUDED__
+
+#include "load_balancing.hpp"
+#include "host.hpp"
+#include "scoped_ptr.hpp"
+
+namespace cass {
+
+class WhitelistPolicy : public ChainedLoadBalancingPolicy {
+public:
+  typedef std::list<std::string> ContactPointList;
```

Comment:
I'd make this a `std::vector<>` or a `std::set<>`. Ideally, this would be a hash table set, but we'd have to either build one or use sparse hash. `std::list<>` is only really good for adding/removing from a large list in the middle of the list.